src/main/java/javaslang/collection/ScanImpl.java

```diff
@@ -0,0 +1,49 @@
+package javaslang.collection;
+
+import java.util.Objects;
+import java.util.function.BiFunction;
+
+class ScanImpl {
```

Comment:
I like the idea of putting it in Traversable, cleaner, but we need a Utils class for that static reverse method. What do you think?